# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 0b191b707750
# Full Hash: 0b191b707750e040289033359306e4966c1bf062
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r 16c5dd2cfd60 -r 0b191b707750 dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Mon Jun 15 18:25:55 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Mon Jun 15 18:25:58 2020 +0000
@@ -100,6 +100,7 @@
 
 namespace layers {
 class CompositableHost;
+class SurfaceDescriptor;
 }
 
 namespace webgl {
@@ -484,9 +485,10 @@
   bool PresentInto(gl::SwapChain& swapChain);
 
  public:
-  void Present();
+  void Present(WebGLFramebuffer*, layers::TextureType);
   RefPtr<gfx::DataSourceSurface> GetFrontBufferSnapshot();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebuffer*,
+                                                  layers::TextureType);
 
   void RunContextLossTimer();
   void CheckForContextLoss();